### ISSUE TITLE
Memory-related error in R 3.2.4

### ISSUE THREAD

**Issue by gwerbin** [wesm/feather#193]
I created a large data frame in Python and saved it using `write_feather`. When I try loading it in R, R crashes with the following error:
```
> results <- feather::read_feather('results.feather')
R(7404,0x7fff78524000) malloc: *** error for object 0x10c4d3630: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
```

It makes no difference whether I use `::` or load the package first.

However, **when I reinstall feather and try to load `results` in the same R session, it works fine**:
```
> install.packages('feather')
trying URL 'https://cloud.r-project.org/bin/macosx/mavericks/contrib/3.2/feather_0.0.1.tgz'
Content type 'application/x-gzip' length 649676 bytes (634 KB)
==================================================
downloaded 634 KB


The downloaded binary packages are in
	/var/folders/z6/mcdmp2d926bd5rh1825k1hnh0000gp/T//RtmpDBxipP/downloaded_packages
> results <- feather::read_feather('results.feather')
There were 15 warnings (use warnings() to see them)
> dim(results)
[1] 526595     50
> results <- feather::read_feather('results.feather')
There were 15 warnings (use warnings() to see them)
> dim(results)
[1] 526595     50
> warnings()
Warning messages:
1: Coercing int64 to double
2: Coercing int64 to double
3: Coercing int64 to double
4: Coercing int64 to double
5: Coercing int64 to double
6: Coercing int64 to double
7: Coercing int64 to double
8: Coercing int64 to double
9: Coercing int64 to double
10: Coercing int64 to double
11: Coercing int64 to double
12: Coercing int64 to double
13: Coercing int64 to double
14: Coercing int64 to double
15: Coercing int64 to double
```

And **when I load a smaller data set first, I am able to load the large data set afterwards** without a problem:

```
> metadata <- feather::read_feather('metadata.feather')
Warning messages:
1: Coercing int64 to double
2: Coercing int64 to double
3: Coercing int64 to double
4: Coercing int64 to double
5: Coercing int64 to double
6: Coercing int64 to double
7: Coercing int64 to double
8: Coercing int64 to double
9: Coercing int64 to double
10: Coercing int64 to double
> dim(metadata)
[1] 2885   60
> results <- feather::read_feather('results.feather')
There were 15 warnings (use warnings() to see them)
> dim(results)
[1] 526595     50
> warnings()
Warning messages:
1: Coercing int64 to double
2: Coercing int64 to double
3: Coercing int64 to double
4: Coercing int64 to double
5: Coercing int64 to double
6: Coercing int64 to double
7: Coercing int64 to double
8: Coercing int64 to double
9: Coercing int64 to double
10: Coercing int64 to double
11: Coercing int64 to double
12: Coercing int64 to double
13: Coercing int64 to double
14: Coercing int64 to double
15: Coercing int64 to double
```

The IRKernel in the Jupyter Notebook crashes hard when trying to load `results.feather`. When trying to load `metadata.feather`, the IRKernel sometimes returns the following error:
```
Error in coldataFeather(x, i): REAL() can only be applied to a 'numeric', not a 'NULL'
```

but I wasn't able to replicate the error in general, and I'm currently using both data sets in the IRKernel without an apparent problem.

I get no errors or crashes in Python operating on this data.

System and package info:

```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.5
BuildVersion:	15F34
```
```
$ R --version
R version 3.2.4 (2016-03-10) -- "Very Secure Dishes"
Copyright (C) 2016 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin13.4.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under the terms of the
GNU General Public License versions 2 or 3.
For more information about these matters see
http://www.gnu.org/licenses/.
```
```
$ Rscript -e "packageVersion('feather')"
[1] ‘0.0.1’
```
```
$ pip show feather-format
---
Metadata-Version: 1.1
Name: feather-format
Version: 0.2.0
Summary: Python interface to the Apache Arrow-based Feather File Format
Home-page: http://github.com/wesm/feather
Author: Wes McKinney
Author-email: wesm@apache.org
License: Apache License, Version 2.0
Location: /usr/local/var/pyenv/versions/rocketrip-datascience/lib/python3.5/site-packages
Requires: cython
Classifiers:
  Development Status :: 3 - Alpha
  Environment :: Console
  Programming Language :: Python
  Programming Language :: Python :: 2
  Programming Language :: Python :: 3
  Programming Language :: Python :: 2.7
  Programming Language :: Python :: 3.4
  Programming Language :: Python :: 3.5
  Programming Language :: Cython
```

**Comment by gwerbin** [comment]
Possibly related to https://github.com/wesm/feather/issues/150 ?

**Comment by wesm** [comment]
How large is the Feather file?

**Comment by RickPack** [comment]
This may be related: R is crashing with an unspecific error message ("R session aborted. R encountered a fatal error. The session was terminated.)" when I attempt to read_feather a 2.5 GB file on an 8 GB RAM machine. Physical memory is about 90% occupied when R crashes. Data.table successfully read a 2.2 GB CSV file even when RAM occupied was 94%. RAM used by the read_feather job increases fast until about 2.5 GB, and then continues increasing slowly until R finally crashes.

**Comment by hadley** [comment]
Have you tried the development version of feather? It's likely to fix this problem.

**Comment by RickPack** [comment]
Hi Hadley - I appreciate all you do and hope you are enjoying celebrity status more than Eminem. 
Yes, I tried the latest version on Github as well as the CRAN version.

**Comment by RickPack** [comment]
I have successfully read a 1.8 GB file so I can easily work around the problem. I have assumed I can ignore this _install_ warning: Warning: this package has a non-empty 'configure.win' file, so building only the main architecture.

**Comment by hadley** [comment]
It's unlikely to be directly a memory problem but instead a bug somewhere in the R code. Is there anyway you could share the file? (Or even better code that generates the file).

**Comment by RickPack** [comment]
I wish I could provide the file or code to produce it, but the critical problem could be that I created this feather file using the feather version from about a month ago. I will aim to create another large file soon and attempt its read.

**Comment by RickPack** [comment]
Success! I read a 2.3GB feather file on my 8 GB ram machine after creating it with the latest development feather version (0.0.1.9000). It was fun to watch the RAM used stay at 99% for a while and then occasionally drop some, and later just a bit, until it rose again. A great deal of RAM became available after the file was read. Thank you for your help, Hadley.

**Comment by wesm** [comment]
Great, looks like this was resolved by the fix for #155. We're planning to put out a new CRAN + Python release in the next week or two (hopefully!)
